src/components/app/xswap/XswapModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { useAccount, useWalletClient } from 'wagmi';
+import { openTransactionModal, renderTxStatus, Transactions } from "@xswap-link/sdk";
+import { useState, useRef, useEffect } from 'react';
+import { waitForTransactionReceipt } from 'viem/actions';
+
+export const XPayButton = () => {
+  const { address, chain, connector } = useAccount();
+  const { data: walletClient } = useWalletClient();
+  const [isHovered, setIsHovered] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  useEffect(() => {
+    return () => {
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+    };
+  }, []);
+
+  const handleOpenXPay = async () => {
+    setIsLoading(true);
+    
+    timeoutRef.current = setTimeout(() => {
+      setIsLoading(false);
+    }, 5000);
+
+    const integratorId = process.env.NEXT_PUBLIC_INTEGRATOR_ID;
+    if (!integratorId) {
+      throw new Error("INTEGRATOR_ID is not set in the .env file");
+    }
+
+    if (!address || !chain || !connector || !walletClient) {
+      throw new Error("Wallet not connected");
+    }
+
+    try {
+      const result = await openTransactionModal({
+        integratorId,
+        dstChain: "42161",
```

Comment:
Does it only support Arbitrum mainnet? how about other chains?
Please also define dstChain and dstToken info as constant variables in the constants folder, if these are constant and not changed in the future.

Reply:
Yes only arbitrum for now.
Good suggestions, i changed them to constant stored in /src/constants/xswap.ts